src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts

```diff
@@ -90,6 +90,10 @@ export class DropdownMenuActionViewItem extends BaseActionViewItem {
 		this._register(this.dropdownMenu.onDidChangeVisibility(visible => {
 			this.element?.setAttribute('aria-expanded', `${visible}`);
 			this._onDidChangeVisibility.fire(visible);
+			if (!visible) {
+				// Wait for the hover to hide before focusing the action item again
```

Comment:
zero does not work here

Reply:
This'll probably still fail on older machines. Can we listen to an event like when the context view hides (context view service?) to make sure we do this after that happens?

Reply:
The context view service fires `onHide` before the view is actually hidden - see here. `onHide` -> `this.onHide()` -> `hide()` -> `onDidChangeVisibility` and the menu is still shown.

![Screenshot 2025-03-28 at 2 38 48 PM](https://github.com/user-attachments/assets/99a2809f-7b52-436f-8a71-ff3e1559c6a1)
![Screenshot 2025-03-28 at 2 38 56 PM](https://github.com/user-attachments/assets/b58b5ca5-8123-44cd-ac81-d615f5d07532)

Reply:
Something's off here, I think that `onHide` is this one:

https://github.com/microsoft/vscode/blob/7c60542555a847fb0d190da819af6fb964949748/src/vs/base/browser/ui/contextview/contextview.ts#L363-L369

Since that's not async, it means `timeout(0)` should be fine to use. But since you mention `timeout(0)` didn't work, something else much be setting focus after it? What is `document.activeElement` after this happened?

Reply:
![Screenshot 2025-03-28 at 2 56 33 PM](https://github.com/user-attachments/assets/55bdf489-88ae-4f68-95ec-f4ae89bbfe1d)

Reply:
I'm wondering if somewhere we should handle `escape` and stop propagation so that focus loss doesn't occur

Reply:
investigating up the call stack 
https://github.com/microsoft/vscode/blob/ff02fe4344632efbb6d6fd44eadf5e183fbfe595/src/vs/base/browser/ui/actionbar/actionbar.ts#L163-L165

Reply:
adding `eventHandled = true;` in the `Escape` case doesn't fix it either...

Reply:
I think the reason `body` is getting focused is because we're calling `focus` on when that element is not yet visible (obscured by the context view)